docs/src/libraries/neutron_ic_querier.md

```diff
@@ -0,0 +1,264 @@
+
+At that point the query is assigned its unique `query_id` identifier, which is not known in advance.
+This identifier is returned to the caller in the reply.
+
+Once the query is registered, the responsible query relayer performs the following steps:
+
+1. fetch the specified value from the target domain
+2. post the query result to `interchainqueries` module
+3. trigger `SudoMsg::KVQueryResult` endpoint on the contract that registered the query
+
+`SudoMsg::KVQueryResult` does not carry back the actual query result. Instead, it posts back
+a `query_id` of the query which had been performed, announcing that its result is available.
+
+That `query_id` can then be used to query the `interchainqueries` module to obtain the raw
+interchainquery result. These raw results fetched from other cosmos chains will be encoded
+in protobuf and require additional processing in order to be reasoned about.
+
+## Library Functions
+
+At its core, this library should support initiating the interchain queries, receiving their
+responses, and reclaiming the escrowed fees by unregistering the queries.
+
+In practice, however, these functions are not very useful in a broader Valence Program context
+by themselves - remote domain *KV-Query* results (next just *V*) arrive back encoded in
+formats meant for those remote domains.
+
+For most cosmos-sdk based chains, *V* is stored in protobuf. Interpreting protobuf from
+within cosmwasm context is not straightforward and requires additional steps.
+Other domains may store their state in other encoding formats. We do not make any assumptions
+about remote domain encodings in this library - instead, that responsibility is handed over
+to the middleware.
+
+For that reason, it is likely that this library will take on the additional responsibility of
+transforming those remote-encoded responses into [canonical data formats](./../middleware/valence_types.md)
+that will be easily recognized within the Valence Protocol scope.
+Aforementioned transformation will be performed by making use of [Valence Middleware](./../middleware/_overview.md).
+
+After the query response is transformed into its canonical representation, the resulting
+data type is written into a [Storage Account](./../components/storage_account.md) making
+it available for further processing, interpretation, or other functions.
+
+## Library Lifecycle
+
+With the baseline functionality in mind, there are multiple design decisions to be made
+that will shape the overall lifecycle of this library.
+
+This library may take on one of two possible routes with respect to how long it is considered
+active:
+1. single-use, instantiated to perform a particular query and complete
+2. multi-use, instantiated to perform multiple domain queries
+
+For single-use design, one of the main questions is at what point does the query
+actually get registered.
+
+One approach for the single-use approach could be to register the query as soon as
```

Comment:
Yes, we should allow the Library to be instantiated with a pre-configured queries.